b2/part.py

```diff
@@ -0,0 +1,41 @@
+######################################################################
+#
+# File: b2/part.py
+#
+# Copyright 2016 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+
+class PartFactory(object):
+    @classmethod
+    def from_list_parts_dict(cls, part_dict):
+        return Part(
+            part_dict['fileId'], part_dict['partNumber'], part_dict['contentLength'],
+            part_dict['contentSha1']
+        )
+
+
+class Part(object):
+    def __init__(self, file_id, part_number, content_length, content_sha1):
+        self.file_id = file_id
+        self.part_number = part_number
+        self.content_length = content_length
+        self.content_sha1 = content_sha1
+
+    def __str__(self):
+        return '<%s %s %s %s %s>' % (
+            self.__class__.__name__, self.file_id, self.part_number, self.content_length,
+            self.content_sha1
+        )
+
+    def __repr__(self):
```

Comment:
if __repr__ is defined, and __str__ is not, the object will behave as though __str__=__repr__. Therefore move the implementation to __repr__ and remove__str__